File: fixtures/scale/julia/medium/src/Registry.jl
module Registry

export registry_add, registry_contains, registry_size

const NAMES = String[]

function registry_add(name::String)
    push!(NAMES, name)
end

function registry_contains(name::String)
    return name in NAMES
end

function registry_size()
    return length(NAMES)
end

end

File: fixtures/scale/r/medium/core/registry.r
# Registry of callable library functions.

.calc_registry <- new.env()
.calc_registry$names <- character(0)

registry_add <- function(name) {
  .calc_registry$names <- c(.calc_registry$names, name)
}

registry_contains <- function(name) {
  name %in% .calc_registry$names
}

registry_size <- function() {
  length(.calc_registry$names)
}
